runtime/common/xwalk_runtime_features.cc

```diff
@@ -11,20 +11,69 @@
 #include "base/logging.h"
 #include "xwalk/runtime/common/xwalk_switches.h"
 
-namespace xwalk {
+namespace {
+
+// Add new features here with the following fields :
+// - Name of the feature
+// - Name of the command line switch which will be used after the
+// --enable/--disable
+// - Description of the feature
+// - Status of the feature : experimental which is turned off by default or
+// stable which is turned on by default
+using Feature = xwalk::XWalkRuntimeFeatures::Feature;
+Feature* g_features[] = {
+  new Feature("SysApps", "sysapps",
+      "Master switch for the SysApps category of APIs", Feature::Stable),
+  new Feature("RawSocketsAPI", "raw-sockets",
+      "JavaScript support for using TCP and UDP sockets", Feature::Stable),
+  new Feature("DeviceCapabilitiesAPI", "device-capabilities",
+      "JavaScript support for peeking at device capabilities", Feature::Stable),
+  new Feature("StorageAPI", "storage",
+      "JavaScript support to file system beyond W3C spec", Feature::Stable),
+  new Feature("ApplicationAPI", "application-api",
+      "JavaScript support for Widget and Manifest APIs", Feature::Stable),
+  new Feature("DialogAPI", "dialog-api",
+      "JavaScript support for dialog APIs", Feature::Experimental)
+};
 
-struct MatchRuntimeFeature
-  : std::unary_function<XWalkRuntimeFeatures::RuntimeFeature, bool> {
-  explicit MatchRuntimeFeature(const std::string& name) : name(name) {}
-  bool operator()(const XWalkRuntimeFeatures::RuntimeFeature& entry) const {
-    return entry.name == name;
-  }
-  const std::string name;
+// Available features on current platform.
+#if defined(OS_ANDROID)
+// FIXME(cmarcelo): The application extensions are currently not fully working
+// for Android, so disable them. See
+// https://crosswalk-project.org/jira/browse/XWALK-674.
+// Android uses a Java extension for device capabilities, so disable the one
+// from sysapps/.
+const char* g_platform_features[] = {
+  "SysApps", "RawSocketsAPI", "StorageAPI", "DialogAPI"
 };
+#elif defined(OS_LINUX_DT)
```

Comment:
don't think we need OS_LINUX_DT, let's just disable bundled features by default (enable for android & tizen)